### DISCUSSION TITLE
Equivalent of "docker run --init" using the Go SDK?

### DISCUSSION THREAD

**Discussion by davechallis** [moby/moby#46883] [Q&A] [answered]
I'm creating some containers in Go using the SDK (https://pkg.go.dev/github.com/docker/docker/client).

I'd like to start a containers with an init process (https://docs.docker.com/engine/reference/run/#specify-an-init-process). On the CLI, I'd do this with `docker run --init ...`, is there an equivalent I can use in the Go API?

**Comment by thaJeztah** [comment, marked as answer]
The `init` option is set in the `HostConfig` struct; see https://github.com/docker/cli/blob/v24.0.7/cli/command/container/opts.go#L673-L676

And https://github.com/moby/moby/blob/df59a357ec01641e7114d500e6a5aa2c4a48f6fc/api/types/container/hostconfig.go#L472-L473

And the API docs; https://docs.docker.com/engine/api/v1.43/#tag/Container/operation/ContainerCreate

**Reply by davechallis** [reply]
Perfect, thank you very much.
